tokio/src/runtime/task/mod.rs

```diff
@@ -361,6 +361,13 @@ impl<S: 'static> Task<S> {
     fn header_ptr(&self) -> NonNull<Header> {
         self.raw.header_ptr()
     }
+
+    cfg_taskdump! {
+        pub(super) fn notify_for_tracing(&self) -> Notified<S> {
+            self.as_raw().state().transition_to_notified_for_tracing();
+            Notified(self.clone())
```

Comment:
Since `transition_to_notified_for_tracing` already increments the refcount, you do not also need to clone it. You probably have a memory leak right now.
```suggestion
            Notified(self)
```

Reply:
I don't think there's a leak here. Remember, when we're calling `notify_for_tracing`, we're iterating over `OwnedTasks`, and need to get notified copies of those tasks into a separate vector. A clone has to occur somewhere. Either we can do it here, or we can modified this method to consume `self` and do the clone in `trace_helper` instead.

Reply:
As far as I can tell, you increment twice (in `transition_to_notified_for_tracing` and `clone`), but only decrement once (in `run`)?

Reply:
Ah, yep. I didn't catch that run `mem::forget`s the `LocalNotified`.

Reply:
Instead of removing the `ref_inc` from `transition_to_notified_for_tracing`, it makes more sense to me to remove the call to `clone`. This way, you only touch the atomic once instead of twice.

Reply:
Done in https://github.com/tokio-rs/tokio/pull/6036/commits/4264f61442b5bb55ece0da6671e3c48b494838d3, but note that there's now a need for cloning in `trace_owned`, because we have to stash the `Task`s (or `Notified`)s outside of `OwnedTasks`.